### PR TITLE
Repository review improvements: observability, perf, validation, CI

### PR DESCRIPTION
**Pull request by Greven145**
## Summary

Full repository review pass implementing 23 backlog items across backend, frontend, infrastructure, and CI/CD.

### Backend (api.web)
- Wire up `AddServiceDefaults()` and `MapDefaultEndpoints()` so health probes (`/health`, `/alive`) respond in all environments — required for ACA liveness/readiness checks
- Enable Application Insights via `UseAzureMonitor()` with `Azure.Monitor.OpenTelemetry.AspNetCore`
- Fix `FactoryClient.GetFactory` swallowing all Cosmos exceptions as 404 — only catches `HttpStatusCode.NotFound` now; 429/503 propagate correctly
- Fix `FindOrSaveAsync` TOCTOU race: catch `DbUpdateException` on conflict and return the existing record instead of throwing
- Pass `CancellationToken` through `ValidateAsync` and `SaveChangesAsync`
- Add array length limits to `FactoryConfigSchemaValidator` (ProductionItems/InputItems/InputResources/AllowedRecipes ≤ 500, NodesPositions ≤ 1000) and validate `Mode` against allowed values

### Frontend (client/)
- Fix `areNodesSame()` stub that always returned `false`, causing spurious re-solves on every node drag
- Replace `import _ from 'lodash'` with `import debounce from 'lodash/debounce'` (tree-shaking)
- Cache the GLPK WASM module instance across solver calls (was reloading on every calculation)
- Replace O(N²) `.find()` scans in LP model-building with `Map` lookups (`objectiveVarMap`, `varsMap`, `adjacencyMap`)
- **Move GLPK solver to a Web Worker** (`solver.worker.ts`) so the main thread stays responsive during calculation — worker bundle emits as a separate chunk

### Infrastructure (AppHost)
- Remove redundant `Aspire.Hosting.NodeJs` package (covered by `Aspire.Hosting.JavaScript`)
- Add `cosmosDb.ConfigureInfrastructure`: set `DefaultTtl = 604800` (7 days) on the factories container
- Add `api.PublishAsAzureContainerApp`: `MinReplicas = 0` (scale to zero), `MaxReplicas = 3`

### Housekeeping
- Remove dead `api/` Azure Functions project from `api.sln` (replaced by `api.web/`)
- Remove credential-logging debug block from `api/Program.cs`
- Fix `client/Dockerfile`: copy path `build/` → `dist/`, env var prefix `REACT_APP_` → `VITE_`, `nginx:latest` → `nginx:1.27-alpine`; add reference-only comments to Dockerfile and `docker-compose.yml`
- Update integration test comments from `// BUG:` → `// FIXED:` for Bugs 1 and 6

### Tests & CI
- Retarget `api.Tests` to `api.web`; fix test data; add 7 new coverage tests for Mode validation and all five array-limit boundaries — **31/31 passing**
- Fix `BrowserFixture` Chromium launch to include `--no-sandbox --disable-dev-shm-usage` for Linux CI runners
- Add MSBuild target to auto-install Playwright browsers after build
- Replace `Task.Delay` timing hacks in Playwright tests with deterministic Playwright waits
- Add `.github/workflows/ci.yml`: two jobs on push/PR to main/master
  - `build-and-test`: restore → build → `dotnet test api.Tests` with TRX upload
  - `build-frontend`: `npm ci` → `npm run build`

## Test plan

- [x] `dotnet test api.Tests` — 31/31 passed
- [x] `dotnet test YetAnotherFactoryPlanner.IntegrationTests` — 11/11 passed (full Aspire stack with Cosmos emulator)
- [x] `cd client && npm run build` — clean Vite build, worker chunk emitted as `solver.worker-*.js`
- [x] `dotnet build api.sln` — 0 errors, 0 warnings

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: api.web/Services/FactoryClient.cs [hunk 1]
```diff
@@ -51,9 +66,9 @@
 
             return factory is not null ? (GetFactoryResult)factory : new None();
         }
-        catch (Exception ex)
+        catch (CosmosException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
         {
-            logger.LogError(ex, "Failed to retrieve factory {FactoryKey}", factoryKey);
+            logger.LogWarning(ex, "Factory {FactoryKey} not found in Cosmos", factoryKey);
```
[the remark this answers is not in the document]
Comment (Greven145) [reply api.web/Services/FactoryClient.cs]: Fixed
